pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/DefaultGroupByExecutor.java

```diff
@@ -183,4 +194,26 @@ public GroupKeyGenerator getGroupKeyGenerator() {
   public GroupByResultHolder[] getGroupByResultHolders() {
     return _groupByResultHolders;
   }
+
+  private Integer getGroupByResultHolderCapacityBasedOnFilterPredicate(QueryContext queryContext) {
+    if (queryContext.getFilter() == null || queryContext.getGroupByExpressions() == null) {
+      return null;
+    }
+
+    List<FilterContext> filterContexts = queryContext.getFilter().getChildren() != null
+        ? queryContext.getFilter().getChildren()
+        : List.of(queryContext.getFilter());
+
+    Map<ExpressionContext, InPredicate> predicateMap = filterContexts.stream()
+        .map(FilterContext::getPredicate)
+        .filter(predicate -> predicate.getType() == Predicate.Type.IN)
```

Comment:
We should also do this for EQ predicate?

Reply:
Added it